homeassistant/components/knx/__init__.py

```diff
@@ -263,6 +272,22 @@ def remove_files(storage_dir: Path, knxkeys_filename: str | None) -> None:
     await hass.async_add_executor_job(remove_files, storage_dir, knxkeys_filename)
 
 
+async def async_remove_config_entry_device(
+    hass: HomeAssistant, config_entry: ConfigEntry, device_entry: DeviceEntry
+) -> bool:
+    """Remove a config entry from a device."""
+    knx_module: KNXModule = hass.data[DOMAIN]
```

Comment:
If you feel extra fancy, maybe you can do https://developers.home-assistant.io/blog/2024/05/01/improved-hass-data-typing (but in a separate PR of course)

Reply:
I do fancy type safety for sure!
Maybe I can even use `runtime_data`, but for now I'm trying to avoid merge conflict hell 😬 (I have some follow-ups ready locally 🙃)